src/marqo/tensor_search/tensor_search.py

```diff
@@ -1906,9 +1943,13 @@ def get_query_vectors_from_jobs(
             merged_vector = np.mean(weighted_vectors, axis=0)
 
             if q.index.normalize_embeddings:
-                norm = np.linalg.norm(merged_vector, axis=-1, keepdims=True)
-                if norm > 0:
-                    merged_vector /= np.linalg.norm(merged_vector, axis=-1, keepdims=True)
+                try:
+                    merged_vector = normalize_vector(merged_vector)
+                except core_exceptions.ZeroMagnitudeVectorError as e:
+                    raise core_exceptions.ZeroMagnitudeVectorError(
+                        f"Magnitude of combined query and context vectors cannot be zero. "
```

Comment:
Agree with it being a convention, however I think trailing space & period make sense here since it's just the first half of the error message. I will however remove the trailing space & period from the second half i.e `If you want to pass a zero vector, please set normalizeEmbeddings = False during index creation. `